h2o-algos/src/main/java/hex/glrm/GLRM.java

```diff
@@ -241,16 +241,26 @@ private void initLoss() {
     for (int i = 0; i < _ncolA; i++) {
       Vec vi = _train.vec(i);
       GlrmLoss lossi = _lossFunc[i];
-      if (vi.isNumeric()) {
+        // take care of pure numeric columns.  Binary numeric columns has isBinary() = true
+      if (vi.isNumeric() && !vi.isBinary()) {
         if (!lossi.isForNumeric())
           error("_loss_by_col", "Loss function " + lossi + " cannot be applied to numeric column " + i);
+      }  else if (vi.isBinary()) {
+        // Allow numeric loss functions, multi-loss on binary columns too; but not the other way round!
+        if (!lossi.isForBinary() && !lossi.isForNumeric() && !lossi.isForCategorical())
+          error("_loss_by_col", "Loss function " + lossi + " cannot be applied to binary column " + i);
+        // make sure we are not calling init mode SVD for categorical binary columns with binary loss
+        if (vi.isCategorical() && lossi.isForBinary()) {
```

Comment:
I don't understand this check. Isn't categorical binary exactly the same as numeric binary (with the exception of the `type` byte)?